db/dcrpg/pgblockchain.go

```diff
@@ -1827,6 +1839,23 @@ func (pgb *ChainDB) AddressTransactionRawDetails(addr string, count, skip int64,
 	return txsRaw, nil
 }
 
+// updateVotesMilestones updates the agenda ID's lockedIn and Activation entries.
+// LockedIn is the height when an agenda is that the agenda passed, but the rules
+// have not yet activated. Activated is the height when the agenda Rule Change
+// is implemented. Find more details on Rule Change Interval (RCI) here:
+// https://docs.decred.org/glossary/#rule-change-interval-rci.
+func updateVotesMilestones(blockChainInfo *dcrjson.GetBlockChainInfoResult) {
+	for agendaID, entry := range blockChainInfo.Deployments {
+		// Add missing agenda ID information.
+		if _, ok := VotingMilestones[agendaID]; !ok && entry.Since > 0 {
+			VotingMilestones[agendaID] = dbtypes.MileStone{
+				LockedIn:  entry.Since,
+				Activated: entry.Since + 8064,
+			}
+		}
+	}
```

Comment:
What if the agenda item is actually "failed" rather than lockedin or active?

Reply:
This makes sense. I will check if the status is "failed" before assigning that activated height value.